### PR TITLE
#163505391 Modularised the codebase, integrated Travis CI and improved unit tests

### PR DESCRIPTION
**Pull request by Oyekunle-Mark**
#### What does this PR do?
Further modularises the codebase, integrates Travis CI into the repo and improves testing.
#### Description of Task to be completed?
* Modularise code
* Integrate Travis CI
* Expand on tests
#### How should this be manually tested?
* ```npm run test```
* Pushing changes to GitHub
#### What are the relevant pivotal tracker stories?
API endpoints should be integrated with Travis CI

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+language: node_js
+node_js:
+  - "stable"
+cache: 
+  directories:
+  - "node_modules"
```
Comment (kenware) [inline .travis.yml]: Add a new line to the end of the file

File: test/apiV1Test.js [hunk 2]
```diff
@@ -12,57 +13,175 @@ describe('POST /parties', function () {
...
+        done(err);
+      });
+  });
+  
+/*
+  it('response should return with status 404 and an error message when a field is not filled', (done) => {
+    request(app)
+      .post('/api/v1/parties')
+      .send({})
+      .set('Accept', 'application/json')
```
Comment (kenware) [inline test/apiV1Test.js]: You can remove this comment if not needed

File: test/apiV1Test.js [hunk 3]
```diff
@@ -12,57 +13,175 @@ describe('POST /parties', function () {
...
+        expect(res.body.data[0]).to.have.property('name');
+        expect(res.body.data[0].logoUrl).to.equal('url');
+        done(err);
+      });
+  });
+
+/*
+  it('respond with a status code 404 and error message when id exceed party numbers', (done) => {
+    request(app)
+      .get('/api/v1/parties/6')
```
Comment (kenware) [inline test/apiV1Test.js]: same here

File: test/apiV1Test.js [hunk 4]
```diff
@@ -71,26 +190,102 @@ describe('POST /offices', function () {
...
+  });
+
+/*
+  it('respond with status code 404 when id exceeds number of party', (done) => {
+    request(app)
+      .get('/api/v1/offices/6')
+      .set('Accept', 'application/json')
+      .expect('Content-Type', /json/)
+      .expect(404)
+      .end((err, res) => {
```
Comment (kenware) [inline test/apiV1Test.js]: same here

**Comment by Oyekunle-Mark** [comment]
Feedback implemented.

**Pull request merged**